.changesets/feat_feature_deduplicate_query_ignored_headers.md

```diff
@@ -0,0 +1,14 @@
+### Add deduplicate_query_ignored_headers option for traffic shaping ([PR #6129](https://github.com/apollographql/router/pull/6129))
+
+When the Router is figuring out if a query can be deduplicated, it considers the entire request, including headers. This means that if you have two requests that have a different header value but are otherwise identical, they will not be deduplicated.
+
+You can use the new `deduplicate_query_ignored_headers` option to provide a list of headers to remove from this deduplication calculation.
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    deduplicate_query: true
+    deduplicate_query_ignored_headers: ["x-my-header"]
+```
```

Comment:
```suggestion
```yaml title="router.yaml"
traffic_shaping:
  all:
    deduplicate_query:
      enabled: true
      ignored_headers:
      - "x-my-header"
```
```

Reply:
I was thinking about that kind of configuration instead

Reply:
I like that too but wouldn't this be a breaking change? 🤔

Reply:
not if a config migration is written. See https://github.com/apollographql/router/tree/dev/apollo-router/src/configuration/migrations for how it is done

Reply:
Oh, awesome. Thanks @Geal / @bnjjj ! Updated with this config and a migration!